data/languages/eecore.sinc

```diff
@@ -959,54 +959,57 @@ with : prime=28 {
         RD128[112,16] = RT128src[112,16] << sa;
     }
     :psllvw RD128, RS128src, RT128src    is microMode=0 & RD128 & RS128src & RT128src & mmiop=0x2 & fct=0x9 {
-        RD128[0,64] = sext(RT128src[0,27] << RS128src[0,5]);
-        RD128[64,64] = sext(RT128src[64,27] << RS128src[64,5]);
+        local tmp:4;
+        tmp = RT128src[0,32] << RS128src[0,5];
+        RD128[0,64] = sext(tmp);
+        tmp = RT128src[64,32] << RS128src[64,5];
+        RD128[64,64] = sext(tmp);
     }
```

Comment:
This change ultimately does the same thing with one exception. The proposed change uses a temporary or local variable which might actually be preferred. May you show a comparison of decompiled results between the two?

Reply:
I tried a couple of test functions of the form 
```
// arguments
 ulong *           a0_lo:4         shifts
 ulong *           a1_lo:4         values
 int *             a2_lo:4         out
// code         
 ld               t0,0x0(a0)
 ld               t1,0x0(a1)
 pextlw           t0,zero,t0
 pextlw           t1,zero,t1
 psllvw           t2,t0,t1
 sq               t2,0x0(a2)
 jr               ra
 _nop
```
but wasn't able to see any difference in decompiled code between the version with temp and the (corrected) version without. Both gave
```
  iVar1 = (int)*values << ((byte)*shifts & 0x1f);
  iVar2 = (int)(*values >> 0x20) << ((byte)(*shifts >> 0x20) & 0x1f);
  *out = iVar1;
  out[1] = iVar1 >> 0x1f;
  out[2] = iVar2;
  out[3] = iVar2 >> 0x1f;
  return;
```
for this example. If you have any suggestions for other uses that might give differences I'm happy to try them out, but otherwise I'll leave it how it was.

Reply:
So decompiling by hand would yield

```c
out[0] = 0;
out[1] = 0;
out[2] = 0;
out[3] = 0;
```

However the result you got would be the worst case scenario since I simplified out the shifts. (0 << x is always 0). Was this a "real life" example you've seen from a game or just one you quickly put together. If it is from a game then it may be a good idea to do specializations for when `RT` is 0. Otherwise I think it is sufficient to just correct the size of the RT bit splice. (ie `RT128src[0,27]` changed to `RT128src[0,32]`).

Did you add specializations for when the shift amount is 0? If not and you have encountered them I agree it would be a good idea to do so.

Reply:
The reason the specializations would need to be done separately is due to a varnode size restriction. For example `RSsrc: rs   	is rs & rs=0      	{ export 0:8; }` is ok but `RSsrc: rs   	is rs & rs=0      	{ export 0:16; }` results in an error.

Reply:
No, this was just an example I made up. My comment about the shift=0 behavior was just a straightforward way to see that this was the right fix.